vllm/distributed/parallel_state.py

```diff
@@ -867,9 +898,12 @@ def init_distributed_environment(
         else:
             local_rank = rank
     global _WORLD
+    global _ADAG_WORLD
     if _WORLD is None:
         ranks = list(range(torch.distributed.get_world_size()))
         _WORLD = init_world_group(ranks, local_rank, backend)
+        _ADAG_WORLD = AdagNcclGroup(_WORLD, workers)
```

Comment:
is this part necessary? is there any way to just pass existing group to .experimental_compile?

Reply:
There needs to be some contract for the communicator used in aDAG. An interface would capture that contract. We  directly pass in existing group, but it could fail at runtime if the contract is broken, which is less desired than enforcement at compile time.